docs/rules/no-import-assign.md

```diff
@@ -0,0 +1,44 @@
+# disallow assigning to imported bindings (no-import-assign)
+
+The updates of imported bindings by ES Modules cause runtime errors.
+
+## Rule Details
+
+This rule warns the assignments, increments, and decrements of imported bindings.
+
+Examples of **incorrect** code for this rule:
+
+```js
+/*eslint no-import-assign: "error"*/
+
+import mod, { named } from "./mod.mjs"
+import * as mod_ns from "./mod.mjs"
+
+mod = 1          // ERROR: 'mod' is readonly.
+named = 2        // ERROR: 'named' is readonly.
+mod_ns.named = 3 // ERROR: the members of 'mod_ns' is readonly.
```

Comment:
what about `Object.assign, `Object.defineProperty`, `Object.defineProperties`, `Reflect.defineProperty`, `Reflect.deleteProperty`, or `Reflect.set` with `mod_ns` as the first argument? All of those attempt (and would fail) to mutate the module namespace object.

Reply:
Good catch. I will add it later.

Reply:
These are also attempting to mutate `mod_ns`:

```js
for (mod_ns.named in foo);
for (mod_ns.named of foo);
[mod_ns.named] = foo;
[...mod_ns.named] = foo;
({ bar:  mod_ns.named } = foo);
({ ...mod_ns.named } = foo);
```

Reply:
@mdjermanovic none of those do any mutations.

Reply:
> @mdjermanovic none of those do any mutations.

I could be wrong, but I think that any of those would try to modify/create property `named` on object `mod_ns` (except for `for-in` and `for-of` when `foo` is empty)?

Reply:
@mdjermanovic Good catch!

Reply:
How could that be the case? All of those are reading values, nothing is assigning to the mod_ns object.

Reply:
@ljharb all members of `mod_ns` (namespace object) are read-only.

Reply:
ah nvm, i missed that they’re reassigning (not mutating) mod_ns, i was confused by the terminology.